### DISCUSSION TITLE
Please confirm if zipkin and/or jaeger receiver works to pull traces into Skywalking.

### DISCUSSION THREAD

**Discussion by lstephe** [apache/skywalking#8994] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

Please confirm if zipkin and/or jaeger receiver works to pull traces into Skywalking.   We have not been successful with either one.  If you can please share some suggestions of how to pull traces into Skywalking.  Thanks

### What you expected to happen

I expected to be able to pull traces into Skywalking.

### How to reproduce

Yes, I am able to reproduce.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by Superskyyy** [comment]
@lstephe hello, thanks for the question. However, please do not label an issue as "bug" until it is really one after discussions. 

Additionally, the subsection for "how to reproduce" is not properly filled, providing "yes we can reproduce" is definitely not helpful for anyone to see the cause. Please update the details. 

Maybe you will be interested in this one as the community is planning to fully support Zipkin traces in near future. https://github.com/apache/skywalking/issues/8932

**Comment by wu-sheng** [comment]
The receiver is for accepting push data, no pull mechanism because considering the size of traces dataset, it is unrealistic to pull.
Same as @Superskyyy shared, we are playing as an alternative option for Zipkin server.

**Comment by lstephe** [comment]
What is the alternative option for Zipkin server, will it be able to pull traces?

Thanks


From: 吴晟 Wu Sheng ***@***.***>
Sent: Thursday, May 5, 2022 7:27 PM
To: apache/skywalking ***@***.***>
Cc: Stephens, Linda ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [apache/skywalking] Please confirm if zipkin and/or jaeger receiver works to pull traces into Skywalking. (Discussion #8994)


*EXTERNAL SENDER*


The receiver is for accepting push data, no pull mechanism because considering the size of traces dataset, it is unrealistic to pull.
Same as @Superskyyy<https://urldefense.com/v3/__https:/github.com/Superskyyy__;!!OUOUp-U!6Zufj8GNyopRLhAM5Hxn0L5Ac8rQeX0yXWhhqgGp2nShqbbNW5NRukKu21Ufi6nLi87KQY5_gJLTYao3mDxsGWerSJc$> shared, we are playing as an alternative option for Zipkin server.

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/apache/skywalking/discussions/8994*discussioncomment-2697361__;Iw!!OUOUp-U!6Zufj8GNyopRLhAM5Hxn0L5Ac8rQeX0yXWhhqgGp2nShqbbNW5NRukKu21Ufi6nLi87KQY5_gJLTYao3mDxs6kQmgQs$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AY7QPXKOLE3NSZDOG6PTHA3VIRKM7ANCNFSM5VGPSPDA__;!!OUOUp-U!6Zufj8GNyopRLhAM5Hxn0L5Ac8rQeX0yXWhhqgGp2nShqbbNW5NRukKu21Ufi6nLi87KQY5_gJLTYao3mDxs-za3pJo$>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

________________________________
NOTICE: All information in and attached to the e-mails below may be proprietary, confidential, privileged and otherwise protected from improper or erroneous disclosure. If you are not the sender's intended recipient, you are not authorized to intercept, read, print, retain, copy, forward, or disseminate this message. If you have erroneously received this communication, please notify the sender immediately by phone (704-758-1000) or by e-mail and destroy all copies of this message electronic, paper, or otherwise. By transmitting documents via this email: Users, Customers, Suppliers and Vendors collectively acknowledge and agree the transmittal of information via email is voluntary, is offered as a convenience, and is not a secured method of communication; Not to transmit any payment information E.G. credit card, debit card, checking account, wire transfer information, passwords, or sensitive and personal information E.G. Driver's license, DOB, social security, or any other information the user wishes to remain confidential; To transmit only non-confidential information such as plans, pictures and drawings and to assume all risk and liability for and indemnify Lowe's from any claims, losses or damages that may arise from the transmittal of documents or including non-confidential information in the body of an email transmittal. Thank you.

**Comment by lstephe** [comment]
Yes, push

From: 吴晟 Wu Sheng ***@***.***>
Sent: Thursday, May 5, 2022 7:00 PM
To: apache/skywalking ***@***.***>
Cc: Stephens, Linda ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [apache/skywalking] Please confirm if zipkin and/or jaeger receiver works to pull traces into Skywalking. (Discussion #8994)


*EXTERNAL SENDER*


I am not sure what do you mean pull traces into skywalking? do you mean push?

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/apache/skywalking/discussions/8994*discussioncomment-2697299__;Iw!!OUOUp-U!_W2k3XPDw7kX_IHUFpL-40Kx1ZJVEnCddhDPe29tLaPrc1HvGyWQRIfNKTe2B16CmtAxGIorY942ehglAR0x-lkHk0I$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AY7QPXJEXBQ672HTQKQCNYLVIRHGDANCNFSM5VGPSPDA__;!!OUOUp-U!_W2k3XPDw7kX_IHUFpL-40Kx1ZJVEnCddhDPe29tLaPrc1HvGyWQRIfNKTe2B16CmtAxGIorY942ehglAR0xwkSbdM4$>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

________________________________
NOTICE: All information in and attached to the e-mails below may be proprietary, confidential, privileged and otherwise protected from improper or erroneous disclosure. If you are not the sender's intended recipient, you are not authorized to intercept, read, print, retain, copy, forward, or disseminate this message. If you have erroneously received this communication, please notify the sender immediately by phone (704-758-1000) or by e-mail and destroy all copies of this message electronic, paper, or otherwise. By transmitting documents via this email: Users, Customers, Suppliers and Vendors collectively acknowledge and agree the transmittal of information via email is voluntary, is offered as a convenience, and is not a secured method of communication; Not to transmit any payment information E.G. credit card, debit card, checking account, wire transfer information, passwords, or sensitive and personal information E.G. Driver's license, DOB, social security, or any other information the user wishes to remain confidential; To transmit only non-confidential information such as plans, pictures and drawings and to assume all risk and liability for and indemnify Lowe's from any claims, losses or damages that may arise from the transmittal of documents or including non-confidential information in the body of an email transmittal. Thank you.

**Comment by lstephe** [comment]
Push is exactly what we are trying to do, but the zipkin receiver is not working.  We are getting the following error:
org.apache.skywalking.oap.server.starter.OAPServerBootstrap 60 [main] ERROR [] - [8.9.1-91ab0a3 (20211210154203)] [receiver_zipkin] missing.4org.apache.skywalking.oap.server.library.module.ModuleNotFoundException: [receiver_zipkin] missing.3at org.apache.skywalking.oap.server.library.module.ModuleManager.init(ModuleManager.java:55) ~[library-module-8.9.1.jar:8.9.1]2at org.apache.skywalking.oap.server.starter.OAPServerBootstrap.start(OAPServerBootstrap.java:44) [server-starter-8.9.1.jar:8.9.1]1at org.apache.skywalking.oap.server.starter.OAPServerStartUp.main(OAPServerStartUp.java:23) [server-starter-8.9.1.jar:8.9.1]


From: 吴晟 Wu Sheng ***@***.***>
Sent: Thursday, May 5, 2022 7:27 PM
To: apache/skywalking ***@***.***>
Cc: Stephens, Linda ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [apache/skywalking] Please confirm if zipkin and/or jaeger receiver works to pull traces into Skywalking. (Discussion #8994)


*EXTERNAL SENDER*


The receiver is for accepting push data, no pull mechanism because considering the size of traces dataset, it is unrealistic to pull.
Same as @Superskyyy<https://urldefense.com/v3/__https:/github.com/Superskyyy__;!!OUOUp-U!6Zufj8GNyopRLhAM5Hxn0L5Ac8rQeX0yXWhhqgGp2nShqbbNW5NRukKu21Ufi6nLi87KQY5_gJLTYao3mDxsGWerSJc$> shared, we are playing as an alternative option for Zipkin server.

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/apache/skywalking/discussions/8994*discussioncomment-2697361__;Iw!!OUOUp-U!6Zufj8GNyopRLhAM5Hxn0L5Ac8rQeX0yXWhhqgGp2nShqbbNW5NRukKu21Ufi6nLi87KQY5_gJLTYao3mDxs6kQmgQs$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AY7QPXKOLE3NSZDOG6PTHA3VIRKM7ANCNFSM5VGPSPDA__;!!OUOUp-U!6Zufj8GNyopRLhAM5Hxn0L5Ac8rQeX0yXWhhqgGp2nShqbbNW5NRukKu21Ufi6nLi87KQY5_gJLTYao3mDxs-za3pJo$>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

________________________________
NOTICE: All information in and attached to the e-mails below may be proprietary, confidential, privileged and otherwise protected from improper or erroneous disclosure. If you are not the sender's intended recipient, you are not authorized to intercept, read, print, retain, copy, forward, or disseminate this message. If you have erroneously received this communication, please notify the sender immediately by phone (704-758-1000) or by e-mail and destroy all copies of this message electronic, paper, or otherwise. By transmitting documents via this email: Users, Customers, Suppliers and Vendors collectively acknowledge and agree the transmittal of information via email is voluntary, is offered as a convenience, and is not a secured method of communication; Not to transmit any payment information E.G. credit card, debit card, checking account, wire transfer information, passwords, or sensitive and personal information E.G. Driver's license, DOB, social security, or any other information the user wishes to remain confidential; To transmit only non-confidential information such as plans, pictures and drawings and to assume all risk and liability for and indemnify Lowe's from any claims, losses or damages that may arise from the transmittal of documents or including non-confidential information in the body of an email transmittal. Thank you.

**Reply by Superskyyy** [reply]
@lstephe  Now I see the problem but seems like it was solved in 8.9.0 https://github.com/apache/skywalking/issues/7842 already, do you have a `zipkin-receiver-plugin.jar` in the oap-libs folder?

**Reply by Superskyyy** [reply]
![image](https://user-images.githubusercontent.com/26076517/167163598-4dc08925-41cd-48ea-8e8a-ad848e710a29.png)

**Comment by Superskyyy** [comment]
Btw, Jaeger receiver was removed back in v8.5.0 since no user feedback, it is no longer supported for now. Are you following an out-dated documentation? Please point out.

**Comment by lstephe** [comment]
Please confirm, the file path where oap-libs folder is located.

Thanks

From: Superskyyy ***@***.***>
Sent: Friday, May 6, 2022 11:20 AM
To: apache/skywalking ***@***.***>
Cc: Stephens, Linda ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [apache/skywalking] Please confirm if zipkin and/or jaeger receiver works to pull traces into Skywalking. (Discussion #8994)


*EXTERNAL SENDER*


@lstephe<https://urldefense.com/v3/__https:/github.com/lstephe__;!!OUOUp-U!-lnPO63ltWM8t4bC6ICHUO4EcMIpEISJqBmfv3hILjW_F79-q8ibKF7k8YkpW7-8bZxi3cgDJyEfaDOm0Xb7W07-FKs$> Now I see the problem but seems like it was solved in 8.9.0 #7842<https://urldefense.com/v3/__https:/github.com/apache/skywalking/issues/7842__;!!OUOUp-U!-lnPO63ltWM8t4bC6ICHUO4EcMIpEISJqBmfv3hILjW_F79-q8ibKF7k8YkpW7-8bZxi3cgDJyEfaDOm0Xb7VWsmypE$> already, do you have a zipkin-receiver-plugin.jar in the oap-libs folder?

-
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/apache/skywalking/discussions/8994*discussioncomment-2701428__;Iw!!OUOUp-U!-lnPO63ltWM8t4bC6ICHUO4EcMIpEISJqBmfv3hILjW_F79-q8ibKF7k8YkpW7-8bZxi3cgDJyEfaDOm0Xb7P42XWQo$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AY7QPXKFIWUAZRL345CGTVLVIU2AHANCNFSM5VGPSPDA__;!!OUOUp-U!-lnPO63ltWM8t4bC6ICHUO4EcMIpEISJqBmfv3hILjW_F79-q8ibKF7k8YkpW7-8bZxi3cgDJyEfaDOm0Xb72wu1qSI$>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

________________________________
NOTICE: All information in and attached to the e-mails below may be proprietary, confidential, privileged and otherwise protected from improper or erroneous disclosure. If you are not the sender's intended recipient, you are not authorized to intercept, read, print, retain, copy, forward, or disseminate this message. If you have erroneously received this communication, please notify the sender immediately by phone (704-758-1000) or by e-mail and destroy all copies of this message electronic, paper, or otherwise. By transmitting documents via this email: Users, Customers, Suppliers and Vendors collectively acknowledge and agree the transmittal of information via email is voluntary, is offered as a convenience, and is not a secured method of communication; Not to transmit any payment information E.G. credit card, debit card, checking account, wire transfer information, passwords, or sensitive and personal information E.G. Driver's license, DOB, social security, or any other information the user wishes to remain confidential; To transmit only non-confidential information such as plans, pictures and drawings and to assume all risk and liability for and indemnify Lowe's from any claims, losses or damages that may arise from the transmittal of documents or including non-confidential information in the body of an email transmittal. Thank you.

**Comment by wu-sheng** [comment]
Too many spams
